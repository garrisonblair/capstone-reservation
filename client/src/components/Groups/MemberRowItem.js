import React, { Component } from 'react';
 import { Button, List } from 'semantic-ui-react';
// import api from '../../utils/api';

class MemberRowItem extends Component {
  state = {
    member:''
  }

  componentDidMount(){
    const { selectedMember } = this.props;
    this.setState({
      member: selectedMember,
    });
  }

  render(){
    const { member } = this.state;
    return(
      <List.Item>
        <List.Content floated='right'>
          <Button>Remove</Button>
        </List.Content>
        <List.Content floated='left'>
          {member}
        </List.Content>
      </List.Item>
    );
  }
}

export default MemberRowItem;
