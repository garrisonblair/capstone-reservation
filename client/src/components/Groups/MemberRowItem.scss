.member-row-item{
    .margin-top{
        margin-top: 5px;
    }
}
