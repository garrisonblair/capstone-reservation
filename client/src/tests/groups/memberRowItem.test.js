import React from 'react';
import renderer from 'react-test-renderer';
import MemberRowItem from '../../components/Groups/MemberRowItem';


it('Shows row with username as admin', () => {
  const member = { username: 'myusername' };
  const tree = renderer
    .create(
      <MemberRowItem
        member={member}
        isAdmin
        deleteFunction={() => { }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Shows row with username as non-admin', () => {
  const member = { username: 'myusername' };
  const tree = renderer
    .create(
      <MemberRowItem
        member={member}
        isAdmin={false}
        deleteFunction={() => { }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
