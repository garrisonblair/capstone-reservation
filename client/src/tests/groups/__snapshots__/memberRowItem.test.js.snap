// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows row with username as admin 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="left floated content"
  >
    <h3>
      <i
        aria-hidden="true"
        className="user icon"
      />
      myusername
    </h3>
  </div>
  <div
    className="right floated content"
  >
    <button
      className="ui button"
      onClick={[Function]}
      role="button"
    >
      Remove
    </button>
  </div>
</div>
`;

exports[`Shows row with username as non-admin 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="left floated content"
  >
    <h3>
      <i
        aria-hidden="true"
        className="user icon"
      />
      myusername
    </h3>
  </div>
  <div
    className="right floated content"
  >
    
  </div>
</div>
`;
